docs/interactive_engine/tutorial_ldbc_gremlin.md

```diff
@@ -831,3 +831,1087 @@ At last, we would like to introduce `endV()` step in path expansion. Since a pat
+                 .in(\'isPartOf\').in(\'isLocatedIn\').in(\'hasMember\').count()')
+print(q1.all())
+```
+
+The output is `[8248]`. It seems to have no problem. However, how many forums are there in the Property Graph?
+
+```python
+# Count how many forums are there in the Property Graph 
+q1 = g.execute('g.V().hasLabel(\'forum\').count()')
+print(q1.all())
+```
+
+ The output is `[8101]`. Here comes the problem: the number of forums having members from India is even larger than the total number of forums. It is impossible! Actually,  when we count how many forums having members from India with previous graph traversal, the last step `.in(\'hasMember\')` will lead to many duplicates, because it is very common for different people to join the same forum. 
+
+Therefore, `dedup()` step is designed to remove the duplicates among traversers. After we add the `dedup()` step before the `count()`, the statistics result is back to normal:
+
+```python
+# Count how may forums have members from India
+q1 = g.execute('g.V().hasLabel(\'place\').has(\'name\', \'India\')\
+                 .in(\'isPartOf\').in(\'isLocatedIn\').in(\'hasMember\').dedup().count()')
+print(q1.all())
+```
+
+The output is `[2822]`. 
+
+For gremlin sentence containing `as()` step, we can also `dedup(TAG)` remove duplicates according to specific tagged entities. For example, the following  gremlin sentence also counts how may forums have members from India, but the traverser starts from `forum`. To remove the duplicates, we `dedup('a')`, where 'a' is the tagged`forum` vertices. 
+
+```python
+# Count how may forums have members from India
+q1 = g.execute('g.V().hasLabel(\'forum\').as(\'a\').out(\'hasMember\')\
+                     .out(\'isLocatedIn\').out(\'isPartOf\').has(\'name\', \'India\')\
+                     .dedup(\'a\').count()')
+print(q1)
+```
+
+The output is also `[2822]`. 
+
+In addition, we can `dedup(TAG1, TAG2, ...)` to remove duplicates by the composition of several tagged entires. For example, the following gremlin sentence counts there are how many different related `(country, forum)` pairs in the Property Graph.
+
+```python
+# Count how many different related (country, forum) pairs in the Property Graph
+q1 = g.execute('g.V().hasLabel(\'place\').as(\'a\')\
+                 .in(\'isPartOf\').in(\'isLocatedIn\')\
+                 .in(\'hasMember\').as(\'b\').dedup(\'a\', \'b\').count()')
+print(q1.all())
+```
+
+The output is `[37164]`
+
+Furthermore, just like previously introduced `where()` step, we can add `by(...)` after `dedup(...)` step to remove duplicates based on the property values. For example, if you want to count there are how many different `firstNames` of `persons` in the Property graph, you can write in GIE as:
+
+```python
+# Count how many different firstName are there in the Property Graph
+q1 = g.execute('g.V().hasLabel(\'person\').dedup().by(\'firstName\').count()')
+print(q1)
+```
+
+The output should be `[432]`.
+
+### Project Steps
```

Comment:
Here, the syntactic sugar of `expr()` can also be used. Cover some examples as well.